“File /nix/store/.../bin/ocamlc, which package ocaml-system.5.1.1 depends upon, is no longer in your path”

Installing a 5.1 switch and then trying to use it, opam complains that the Nix path no longer exists:
```
$ nix-opam-switch create 5.1
$ cd <some project>
$ opam install . --verbose --deps-only
[WARNING] Failed checks on guerre_webapp package definition from source at
          git+file:///home/olivier/hacking_days/2024_01_guerre_sur_le_web/guerre_webapp#main:
    error 23: Missing field 'maintainer'
  warning 25: Missing field 'authors'
  warning 35: Missing field 'homepage'
  warning 36: Missing field 'bug-reports'
  warning 68: Missing field 'license'
[WARNING] File /nix/store/w4zxwfn57qjvv0mi8v0si7bbv1lhdvca-ocaml-5.1.1/bin/ocamlc, which
          package ocaml-system.5.1.1 depends upon, is no longer in your PATH.
[ERROR] Package conflict!
  * Missing dependency:
    - ocaml-system.5.1.1: no longer available

No solution found, exiting
```

Strangely, the file is in the PATH from the normal shell:

```
$ eval $(opam env)

$ opam pin add dune.3.13 https://github.com/ocaml-wasm/dune.git
[NOTE] Package dune is already pinned to git+https://github.com/ocaml-wasm/dune.git (version 3.13).
[dune.3.13] synchronised (no changes)
dune is now pinned to git+https://github.com/ocaml-wasm/dune.git (version 3.13)

[WARNING] File /nix/store/w4zxwfn57qjvv0mi8v0si7bbv1lhdvca-ocaml-5.1.1/bin/ocamlc, which package ocaml-system.5.1.1 depends
          upon, is no longer in your PATH.
[ERROR] Package conflict!
  * Missing dependency:
    - ocaml-system.5.1.1: no longer available

[NOTE] Pinning command successful, but your installed packages may be out of sync.

$ which ocamlc
/home/olivier/woo/wasm_of_ocaml/_opam/bin/ocamlc

$ which ocamlc.opt
/home/olivier/woo/wasm_of_ocaml/_opam/bin/ocamlc.opt

$ file $(which ocamlc.opt)
/home/olivier/woo/wasm_of_ocaml/_opam/bin/ocamlc.opt: symbolic link to /nix/store/w4zxwfn57qjvv0mi8v0si7bbv1lhdvca-ocaml-5.1.1/bin/ocamlc.opt

$ file /nix/store/w4zxwfn57qjvv0mi8v0si7bbv1lhdvca-ocaml-5.1.1/bin/ocamlc.opt
/nix/store/w4zxwfn57qjvv0mi8v0si7bbv1lhdvca-ocaml-5.1.1/bin/ocamlc.opt: ELF 64-bit LSB executable, x86-64, version 1 (SYSV), dynamically linked, interpreter /nix/store/9y8pmvk8gdwwznmkzxa6pwyah52xy3nk-glibc-2.38-27/lib/ld-linux-x86-64.so.2, for GNU/Linux 3.10.0, not stripped

$ file /nix/store/w4zxwfn57qjvv0mi8v0si7bbv1lhdvca-ocaml-5.1.1/bin/ocamlc    
/nix/store/w4zxwfn57qjvv0mi8v0si7bbv1lhdvca-ocaml-5.1.1/bin/ocamlc: symbolic link to ocamlc.opt
```

--- a/nix-opam-switch
+++ b/nix-opam-switch
@@ -91,6 +91,9 @@ case "$cmd" in
     prefix=$(opam var --switch "$switch_name" prefix)
     # Copy the compiler and tools into the created switch
     cp -rd --no-preserve=mode -T "$out" "$prefix"
+    # Hack: Symlinks interfere with Opam's system package integrity check
+    cp -L --remove-destination "$out/bin/ocamlc"{,.opt} "$prefix/bin/"
+    chmod +x "$prefix/bin/ocamlc"{,.opt}
     # Register a GC root to keep $out from being garbage collected
     nix-store --add-root "$prefix/nix-roots/switch" -r "$out" >/dev/null
     ;;

